src/test/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibraryUnitTest.java

```diff
@@ -19,9 +18,170 @@
+    }
+
+
+    /**
+     * Composes the Platform unit name for the read group given its index.
+     *
+     * @param index the index of the targeted read group in the range
+     *              [1,{@link #groupCount}]
+     * @return never <code>null</code>.
+     */
+    private String composePlatformUnitName(final int index) {
+        checkReadGroupIndex(index);
+        return DEFAULT_PLATFORM_UNIT_PREFIX + (((index-1)%3)+1);
+    }
+
+    /**
+     * Checks the correctness of a given read group index.
+     *
+     * A correct index is any value in the range [1,{@link #groupCount}].
+     *
+     * @param index the target index.
+     * @throws IllegalArgumentException if the input index is not correct.
+     */
+    private void checkReadGroupIndex(final int index) {
+        if (index < 1 || index > groupCount) {
+            throw new IllegalArgumentException(
+                    "read group" + " index ("
+                            + index
+                            + ") is out of bounds [1," + groupCount + "]");
+        }
+    }
+
+    /**
+     * Returns the ID for a read group given its index.
+     *
+     * @param index the index of the targeted read group in the range
+     *              [1,{@link #groupCount}]
+     * @return never <code>null</code> and must be unique to each
+     *              possible read group index.
+     */
+    private String getReadGroupId(final int index) {
+        checkReadGroupIndex(index);
+        return getHeader().getReadGroups().get(index - 1).getReadGroupId();
+    }
+
+    @BeforeClass
+    public void beforeClass() {
+        header = ArtificialSAMUtils.createArtificialSamHeader(
+                DEFAULT_CHROMOSOME_COUNT, DEFAULT_CHROMOSOME_START_INDEX,
+                DEFAULT_CHROMOSOME_SIZE);
+
+        final List<String> readGroupIDs = new ArrayList<>();
+        final List<String> sampleNames = new ArrayList<>();
+
+        for (int i = 1; i <= groupCount; i++) {
+            final String readGroupId = composeReadGroupId(i);
+            readGroupIDs.add(readGroupId);
+            sampleNames.add(readGroupId);
+        }
+
+        ArtificialSAMUtils.createEnumeratedReadGroups(
+                header, readGroupIDs, sampleNames);
+
+        for (int i = 1; i <= groupCount; i++) {
+            final String readGroupId = readGroupIDs.get(i-1);
+            final SAMReadGroupRecord groupRecord = header.getReadGroup(readGroupId);
+            groupRecord.setAttribute("PL", composePlatformName(i));
+            groupRecord.setAttribute("PU", composePlatformUnitName(i));
+        }
+
+    }
+
+    @AfterClass
+    public void afterClass() {
+        header = null;
+    }
+
+    /**
+     * Creates a read record.
+     *
+     * @param cigar the new record CIGAR.
+     * @param group the new record group index that must be in the range \
+     *              [1,{@link #groupCount}]
+     * @param reference the reference sequence index (0-based)
```

Comment:
it is 0 based. This is the reference index (in the dictionary). This is *not* the index in the reference.